node/agent.go

```diff
@@ -407,8 +407,8 @@ func (a *HostAgent) Start(shutdown <-chan interface{}) {
 		case <-startExit:
 			glog.Infof("Host Agent restarting")
 			close(unregister)
-			unregister = make(chan interface{})
 			rwg.Wait()
+			unregister = make(chan interface{})
```

Comment:
We close unregister channel and then wait for the goroutine on line 366 to exit.  If we reassign the unregister variable to a fresh channel at the wrong time, the goroutine on 366 never exits. If we happen to close and then reassign unregister while the RegisterHost call on line 371 is executing, AND that call returns an error, then we will loop back into 371 passing in an unclosed channel that will never get closed.